src/data_provider/src/sysInfoMac.cpp

```diff
@@ -170,16 +173,22 @@ std::string SysInfo::getSerialNumber() const
 
 static void getPackagesFromPath(const std::string& pkgDirectory, const int pkgType, std::function<void(nlohmann::json&)> callback)
 {
-    const auto packages { Utils::enumerateDir(pkgDirectory) };
-
-    for (const auto& package : packages)
+    if (MACPORTS == pkgType)
     {
-        if (PKG == pkgType)
+        if (Utils::existsRegular(pkgDirectory + "/" + MACPORTS_DB_NAME))
         {
-            if (Utils::endsWith(package, ".app"))
+            std::shared_ptr<SQLite::IConnection> sqliteConnection = std::make_shared<SQLite::Connection>(pkgDirectory + "/" + MACPORTS_DB_NAME);
+            SQLite::Statement stmt
+            {
+                sqliteConnection,
+                MACPORTS_QUERY
+            };
+
+            while (SQLITE_ROW == stmt.step())
             {
                 nlohmann::json jsPackage;
-                FactoryPackageFamilyCreator<OSType::BSDBASED>::create(std::make_pair(PackageContext{pkgDirectory, package, ""}, pkgType))->buildPackageData(jsPackage);
+                std::pair<SQLite::Statement, int> pkgContext {std::make_pair(stmt, pkgType)};
```

Comment:
I think that we are creating a new `pkgContext` variable in every loop cicle, copying `stmt` and `pkgType` every time. 

Maybe we could create `pkgContext` one time outside the loop and the `stmt.step()` instruction will move the data to the next row by itself

```c++
std::pair<const SQLite::Statement &, const int &> pkgContext {std::make_pair(std::cref(stmt), std::cref(pkgType))
```

Reply:
Thanks for the suggestion, I did something similar, but couldn't define SQLite::Statement as constant, since some of the member functions seem to not have the const identifier. For that reason, I use std:ref instead of std::cref.

![2023-02-09_12-33](https://user-images.githubusercontent.com/13010397/217869869-b66e39a9-4608-43b0-bfff-70fef000e91e.png)

Reply:
I make suggestions, but they weren't tested in the real code and they might require adjustments/corrections. 
Thanks to you!